src/js/Aladin.js

```diff
@@ -1437,23 +1445,36 @@ export let Aladin = (function () {
     Aladin.prototype.getViewWCS = function (options) {
         var raDec = this.getRaDec();
         var fov = this.getFov();
-        // TODO: support for other projection methods than SIN
+        var projectionName = this.getProjectionName();
+
+        // get the cootype prefix from the coordinate frame
+        switch (this.getCooFrame()) {
+            case "J2000":
+            case "J2000d":
+                var cooType1 = "RA---";
+                var cooType2 = "DEC--";
+                break;
+            case "Galactic":
+                var cooType1 = "GLON-";
+                var cooType2 = "GLAT-";
+        }
+
         return {
             NAXIS: 2,
             NAXIS1: this.view.width,
             NAXIS2: this.view.height,
-            RADECSYS: 'ICRS',
+            RADECSYS: "ICRS",
             CRPIX1: this.view.width / 2,
             CRPIX2: this.view.height / 2,
             CRVAL1: raDec[0],
```

Comment:
If the frame is *Galactic*, CRVAL1 and CRVAL2 must be given in galactic coordinates.

Reply:
Ah yes!